securedrop_client/logic.py

```diff
@@ -239,20 +238,14 @@ def call_api(self, function, callback, timeout, *args, current_object=None,
         self.api_threads[new_thread_id] = {
             'thread': new_api_thread,
             'runner': new_api_runner,
-            'timer': new_timer,
         }
 
         # Start the thread and related activity.
         new_api_thread.start()
 
-    def clean_thread(self, thread_id):
-        """
-        Clean up the identified thread's state after an API call.
-        """
-        if thread_id in self.api_threads:
-            timer = self.api_threads[thread_id]['timer']
-            timer.disconnect()
-            del(self.api_threads[thread_id])
+    def on_api_timeout(self) -> None:
```

Comment:
Looks good. This is where we can add retry logic sometime after this PR.

Reply:
Also if all the retries fail, then perhaps we will want to emit another signal so we know the type of message that will need to be displayed to the user in the gui.

For now, do you imagine that the gui will listen for the call_timed_out signal or that the failure callbacks will figure out what kind of exception occurred?

Reply:
There is only one type of exception that is a timeout (`RequestTimeoutError`), and that is handled via this function and this line

```python
new_api_runner.call_timed_out.connect(self.on_api_timeout)
```